pkg/controller/util/constants.go

```diff
@@ -26,6 +26,7 @@ const (
 
 	NamespaceKind     = "Namespace"
 	ServiceKind       = "Service"
+	ClusterNameField  = "clusterName"
```

Comment:
This is good, but I still have some concerns if we do not sync up the YAML templates for example and test.

1) People add a new federated resource, and add some test cases YAML template with `override` policy, and use correct `clusterName` in YAML, CI test passed.
2) In example file, they use `clustername`, but this will not be detected if people do not test the example in a federation environment.

If we can keep the YAML templates in example and test keep synced, then we will not have such issues. Comments @marun

Reply:
I agree that the walkthrough needs to be tested, but I think that requirement is orthogonal to other forms of testing.  #104 is intended to track the need to automate the walkthrough so it can be tested in CI (cc: @font).  Also, #60 (my next work item) will remove the need to define primitives in code (they will exist only in the api).